cmd/scollector/collectors/cpu_windows.go

```diff
@@ -15,35 +15,65 @@ func init() {
 
 func c_cpu_windows() (opentsdb.MultiDataPoint, error) {
 	var dst []Win32_PerfRawData_PerfOS_Processor
-	var q = wmi.CreateQuery(&dst, `WHERE Name <> '_Total'`)
+	var q = wmi.CreateQuery(&dst, "")
 	err := queryWmi(q, &dst)
 	if err != nil {
 		return nil, err
 	}
 	var md opentsdb.MultiDataPoint
 	var used, num uint64
+	var winCPUTotal *Win32_PerfRawData_PerfOS_Processor
 	for _, v := range dst {
+		if v.Name == "_Total" {
+			winCPUTotal = &v
+			continue
+		}
 		ts := TSys100NStoEpoch(v.Timestamp_Sys100NS)
+		metricName := "win.cpu"
 		num++
 		//Divide by 1e5 because: 1 seconds / 100 Nanoseconds = 1e7. This is the percent time as a decimal, so divide by two less zeros to make it the same as the result * 100.
 		used += (v.PercentUserTime + v.PercentPrivilegedTime + v.PercentInterruptTime) / 1e5
-		AddTS(&md, "win.cpu", ts, v.PercentPrivilegedTime/1e5, opentsdb.TagSet{"cpu": v.Name, "type": "privileged"}, metadata.Counter, metadata.Pct, "Percentage of non-idle processor time spent in privileged mode.")
-		AddTS(&md, "win.cpu", ts, v.PercentInterruptTime/1e5, opentsdb.TagSet{"cpu": v.Name, "type": "interrupt"}, metadata.Counter, metadata.Pct, "Percentage of time that the processor spent receiving and servicing hardware interrupts during the sample interval.")
-		AddTS(&md, "win.cpu", ts, v.PercentUserTime/1e5, opentsdb.TagSet{"cpu": v.Name, "type": "user"}, metadata.Counter, metadata.Pct, "Percentage of non-idle processor time spent in user mode.")
-		AddTS(&md, "win.cpu", ts, v.PercentIdleTime/1e5, opentsdb.TagSet{"cpu": v.Name, "type": "idle"}, metadata.Counter, metadata.Pct, "Percentage of time during the sample interval that the processor was idle.")
-		AddTS(&md, "win.cpu.interrupts", ts, v.InterruptsPersec/1e5, opentsdb.TagSet{"cpu": v.Name}, metadata.Counter, metadata.Event, "Average number of hardware interrupts that the processor is receiving and servicing in each second.")
-		Add(&md, "win.cpu.dpcs", v.DPCRate, opentsdb.TagSet{"cpu": v.Name}, metadata.Counter, metadata.Event, "Rate at which deferred procedure calls (DPCs) are added to the processor DPC queue between the timer tics of the processor clock.")
-		AddTS(&md, "win.cpu.time_cstate", ts, v.PercentC1Time/1e5, opentsdb.TagSet{"cpu": v.Name, "type": "c1"}, metadata.Counter, metadata.Pct, "Percentage of time that the processor spends in the C1 low-power idle state, which is a subset of the total processor idle time.")
```

Comment:
Also there are actually like 7 CState levels, so we really just have a high level view anyway. If you  see something strange in the totals you would need to use a more detailed tool to troubleshoot further (or you just turn them off globally like we did)